### PR TITLE
Refactor and cleanup GLSPContribution API

### PR DESCRIPTION
**Pull request by tortmayr**
- Refactor `GLSPCLientProvider` and reuse it in `BaseGLSPTheiaConnector`
- Make GLSPContributions disposable
- Cleanup redundant session concept in `GLSPBackendContribution`
- Cleanup & refactor`GLSPClientContribution` API.
- Cleanup & refactor`GLSPSocketServerContribution` API
- Remove remains from @theia/languages
- Simplify client initialization by using Theia's Deferred concept.

Fixes https://github.com/eclipse-glsp/glsp/issues/848

### CODE REVIEW

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Changes look good to me and seem to work well. I have a few questions regarding code which I'd like to clarify first.

File: packages/theia-integration/src/browser/glsp-client-contribution.ts [hunk 1]
```diff
@@ -83,84 +109,58 @@ export abstract class BaseGLSPClientContribution implements GLSPClientContributi
...
-                )
-            ]);
-        }
-        return this.ready;
-    }
+    waitForActivation?(app: FrontendApplication): Promise<void>;
 
-    activate(): Disposable {
-        if (this.toDeactivate.disposed) {
+    activate(app: FrontendApplication): void {
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/theia-integration/src/browser/glsp-client-contribution.ts:114]: => MaybePromise

**Comment by tortmayr** [comment]
Thanks for the fast review, I have added a commit that should address the open issues

**Review by martin-fleck-at — APPROVED** [review]
LGTM!

**Pull request merged**